### DISCUSSION TITLE
Problem where only the features of the edges are important.

### DISCUSSION THREAD

**Discussion by MehdiZouitine** [pyg-team/pytorch_geometric#4971] [Q&A]
Hello,

I've been stuck with this problem for a few days.
I have a graph where only the weights of the edges are important (TSP with the cost of the edges between each city).
Is there a GNN layer that can learn a good node representation from the edges only?
<img width="434" alt="image" src="https://user-images.githubusercontent.com/39647484/178578733-e8812aa7-d5d0-4449-a3af-7edef0e7f58e.png">

Another simple problem that I can't solve: 
Knowing a graph with weighted edges : train a GNN to predict the total weight of the graph.

I have tried several types of graph convolution but it doesn't work...

Thanks you very much

**Comment by Padarn** [comment]
> Knowing a graph with weighted edges : train a GNN to predict the total weight of the graph.

Can't you just sum the edge weights?

**Reply by Padarn** [reply]
> Is there a GNN layer that can learn a good node representation from the edges only?

Not totally sure what 'good' means here. But maybe this helps, it is the implementation of shorted path problem using GNN (in the `graph_nets` repo):
https://colab.research.google.com/github/deepmind/graph_nets/blob/master/graph_nets/demos/shortest_path.ipynb

There are a number of papers that (try to) solve this problem, just sharing this one as they have a collab you can explore.

**Reply by MehdiZouitine** [reply]
That is not the issue. Of course I can do the sum and not use GNN because the problem is trivial.
I would just like to know if a GNN is able to handle a graph with only edge features.

I will take look on the repo !

**Reply by Padarn** [reply]
I'm not clear on your weight edge sum problem then sorry

**Comment by MehdiZouitine** [comment]
Do you know how to handle it ?

**Reply by Padarn** [reply]
Sorry I do not understand what you want to achieve so I can't help. Maybe someone else will be able to help you.

**Reply by rusty1s** [reply]
I think you should try `GraphConv` with `aggr=sum`, and pass `edge_weight` to its `forward` call as well.
